mantidproject/mantid
# Adopt Transfit resonance fitting algorithm

**Description of work.**
Onboard a new PEARL specific fit function and algorithm to make use of it from Chris Ridley. The fit function is in PEARLTransVoight.py, and the algorithm in PEARLTransfit.py. The aim of the Transfit algorithm is to fit a Voigt function to neutron resonances in order to determine the temperature of a sample.

Original files and documentation from Chris:
[ResonanceFitting (3).zip](https://github.com/mantidproject/mantid/files/5334865/ResonanceFitting.3.zip)

**Report to:** Chris Ridley

**To test:**
1. Ensure that archive access is enabled
2. Test for single run: Execute the PEARLTransfit algorithm to open the dialogue
3. Enter a PEARL run, e.g. PEARL00073987, set Calibrate to True and run the algorithm
4. Assuming calibration workspaces were successfully made, run the algorithm again with Calibration disabled
5. Check that the `T` workspaces were successfully made and that the sample temperature information was output in the message pane
6. Test for multiple runs: repeat, but this time with several runs e.g. PEARL00073987-00073990 as Files input
This should take the average of the workspaces and use it for the fit

Fixes #29067 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
